core/trino-spi/src/main/java/io/trino/spi/block/DictionaryBlock.java

```diff
@@ -87,6 +87,8 @@ private static Block createInternal(int positionCount, Block dictionary, int[] i
             }
             dictionary = dictionaryBlock.getDictionary();
             ids = newIds;
+            // IDs changed so dictionary source id needs to change
+            dictionarySourceId = randomDictionaryId();
```

Comment:
Can't this be the sourceId of the underlying dictionary? (i.e., `dictionaryBlock.getDictionarySourceId()`) ?